spec/folder_spec.rb

```diff
@@ -147,13 +147,16 @@ def post_create_folder path, params={}
       expect(json_body[:error]).to eq('Invalid folder: "wisdom.txt"')
     end
 
-    it 'refuses to create existing folder' do
+    it 'creating an existing folder is idempotent' do
       token_header(:editor)
       blueprints_folder = create_folder('athena', 'blueprints')
       post_create_folder('blueprints')
 
-      expect(last_response.status).to eq(422)
-      expect(json_body[:error]).to eq('Folder exists')
+      expect do
+        expect(last_response.status).to eq(200)
```

Comment:
Just curious if this behavior changes any UX-related issue on the UI side? I can imagine the `422` previously would show up in the error message area for the user, but would they not get a message with the `200` and perhaps get confused and not realize that a folder with that name already exists? Or maybe that's okay and they'll find the "old" folder when they look for the "new" folder?

Reply:
So previously you could only create shallow folders anyways.  In the new world, you get a 200 OK where as before you got an error message, but in both cases the user can expect that the folder they created now exists, but minus the unnecessary error message.